rust/noosphere/src/storage.rs

```diff
@@ -82,3 +62,36 @@ fn get_scoped_path(path: &Path, scope: &Did) -> PathBuf {
 
     path_buf
 }
+
+/// Construct [PlatformStorage] from a [StorageLayout] and [StorageConfig].
+///
+/// Takes a [Url] to an IPFS Gateway that is used when compiling with `ipfs-storage`.
+pub async fn create_platform_storage(
+    layout: StorageLayout,
+    #[allow(unused)] ipfs_gateway_url: Option<Url>,
+    #[allow(unused)] storage_config: Option<StorageConfig>,
+) -> Result<PlatformStorage> {
+    #[cfg(any(sled, rocksdb))]
+    let storage = {
+        use noosphere_storage::ConfigurableStorage;
+        let path: PathBuf = layout.into();
+        crate::platform::PrimitiveStorage::open_with_config(
+            &path,
+            storage_config.unwrap_or_default(),
+        )
+        .await?
+    };
+
+    #[cfg(wasm)]
+    let storage = noosphere_storage::IndexedDbStorage::new(&layout.to_string()).await?;
+
+    #[cfg(ipfs_storage)]
+    let storage = {
+        let maybe_client = ipfs_gateway_url.map(|url| GatewayClient::new(url));
+        IpfsStorage::new(storage, maybe_client)
+    };
+
+    debug!("Creating platform storage: {:#?}", storage);
```

Comment:
This log line might be better written before the storage is created (otherwise change to past tense e.g., "created" not "creating"; but, I think moving it is better so that you get appropriate context for any subsequent error).

Reply:
Changed to `created`, as we use `storage`'s Debug implementation to log storage type and extra configurations (could add more debugging to type if needed e.g. `trait StorageInfo { fn name() -> String }`